README.md

```diff
@@ -28,7 +28,373 @@ UpSet: Visualization of Intersecting Sets
+- `provVis` (optional): [Sidebar options](#sidebar-options) for the provenance visualization sidebar. See [Trrack-Vis](https://github.com/Trrack/trrackvis) for more information about Trrack provenance visualization.
+- `elementSidebar` (optional): [Sidebar options](#sidebar-options) for the element visualization sidebar. This sidebar is used for element queries, element selection datatable, and supplimental plot generation.
+- `altTextSidebar` (optional): [Sidebar options](#sidebar-options) for the text description sidebar. This sidebar is used to display the generated text descriptions for an Upset 2.0 plot, given that the `generateAltText` function is provided.
+- `generateAltText` (optional)(`() => Promise<AltText>`): Async function which should return a generated AltText object. See [Alt Text Generation](#alt-text-generation) for more information about Alt Text generation.
+
+##### Configuration (Grammar) options
+
+The configuration (grammar) is used both as internal tracking for UpSet 2.0, and to generate the alt-text via Multinet's API. Other implementations which are attempting to generate text descriptions using the same API must generate a grammar containing the values below. Reference [upset-alt-txt-gen](https://github.com/visdesignlab/upset-alt-txt-gen) for more information about generating text descriptions.
+
+If no configuration options are provided, the default will be:
+
+- No aggregation
+- Sort rows by `Size - Descending`
+- Sort visible sets by `Alphabetical`
+- Minimum degree filter: 0
+- Maximum degree filter: 6
+- Show the "no set" row
+- Hide empty rows
+
+The configuration options are documented below:
+
+- `plotInformation`: An object which contains meta data abut the plot. The three keys are `description`, `sets`, and `items`. The values should be strings.
+  - `description`: A short description about the dataset. Ex: Movie genres and ratings
+  - `sets`: A very short description of what the sets are in the data. Ex: Movie Genres
+  - `items`: A very short description of what the items in the data are. Ex: Movies
+- `horizontal`: Boolean value describing whether or not the plot is horizontal. Defaults to `false`. *Note* this value is not used by UpSet 2.0 but may be used to generate text descriptions.
+- `firstAggregateBy`: Describes the first aggregation state. Defaults to `None`. Possible values:
```

Comment:
This is not a javascript "null" type but rather its a string that's (rather verbosely) saying that there is *No* aggregation. This is editable, but could potentially cause issues with upset-alt-txt. I'd have to review that code to determine if this would be a breaking change.